source/extensions/http/custom_response/local_response_policy/local_response_policy.cc

```diff
@@ -19,19 +20,25 @@ namespace CustomResponse {
 LocalResponsePolicy::LocalResponsePolicy(
     const envoy::extensions::http::custom_response::local_response_policy::v3::LocalResponsePolicy&
         config,
-    Server::Configuration::CommonFactoryContext& context)
+    Server::Configuration::ServerFactoryContext& context)
     : local_body_{config.has_body() ? absl::optional<std::string>(Config::DataSource::read(
                                           config.body(), true, context.api()))
                                     : absl::optional<std::string>{}},
-      formatter_(config.has_body_format()
-                     ? Formatter::SubstitutionFormatStringUtils::fromProtoConfig(
-                           config.body_format(), context)
-                     : nullptr),
       status_code_{config.has_status_code()
                        ? absl::optional<Envoy::Http::Code>(
                              static_cast<Envoy::Http::Code>(config.status_code().value()))
                        : absl::optional<Envoy::Http::Code>{}},
-      header_parser_(Envoy::Router::HeaderParser::configure(config.response_headers_to_add())) {}
+      header_parser_(Envoy::Router::HeaderParser::configure(config.response_headers_to_add())) {
+
+  // TODO(wbpcode): these is a potential bug of message validation. The validation visitor
+  // of server context should not be used here directly. But this is bug is not introduced
+  // by this PR and will be fixed in the future.
+  Server::GenericFactoryContextImpl generic_context(context, context.messageValidationVisitor());
```

Comment:
mm, if this may be a problem shouldn't we not switch to server factory context here?  I'd think if we wanted to inherit validation we'd need to keep this a generic.